### PR TITLE
fix: remove usage of stateful jwt value causing stale closure

### PR DESCRIPTION
**Pull request by jiejasonliu**
### i'm praying that it works this time
---

**tl;dr** stale closures resulted in outdated values for the auth token after refresh

---

Using `useState` gives the state and its updater, if we pass the state to an async function / callback such that a closure is created, the state is captured once. 

Any further updates on it will refer to the now stale value. 

Furthermore, `setState` is async in nature, so there's no guarantee to when the state value actually updates.

A common solution to this is to use `useRef` which is reference-based, but afaik this isn't an easy drop in solution for Recoil.

Instead, I opted to drop the usage of stateful values in the fetch wrapper:
- it's suppose to pure/stateless, we shouldn't have any side effects
- grabbing the value directly from local storage makes more semantic sense

### CODE REVIEW

**Comment by derricklee43** [comment]
looks good to me! This JWT is no fun, lol

**Review by derricklee43 — APPROVED** [review]
